test/test_ourdna_dashboard.py

```diff
@@ -39,124 +39,112 @@ async def setUp(self) -> None:
-                'meta': {
-                    # 'collection-time': '2022-07-03 13:28:00',
-                    'processing-site': 'Garvan',
-                    # 'process-start-time': '2022-07-03 16:28:00',
-                    # 'process-end-time': '2022-07-03 19:28:00',
-                    'received-time': '2022-07-03 14:28:00',
-                    'received-by': 'YP',
-                    'collection-lab': 'XYZ LAB',
-                    'courier': 'ABC COURIERS',
-                    'courier-tracking-number': 'ABCDEF12562',
-                    'courier-scheduled-pickup-time': '2022-07-03 13:28:00',
-                    'courier-actual-pickup-time': '2022-07-03 13:28:00',
-                    'courier-scheduled-dropoff-time': '2022-07-03 13:28:00',
-                    'courier-actual-dropoff-time': '2022-07-03 13:28:00',
-                    'concentration': 1.66,
-                },
-                'type': 'blood',
-                'active': True,
-            },
+        self.participants_external_objects = [
+            participant.to_external() for participant in participants
         ]
 
-        sample_names = ['sample_one', 'sample_two', 'sample_three']
-
         self.sample_external_objects: list[SampleUpsert] = []
         self.sample_internal_objects: list[SampleUpsertInternal] = []
 
-        for sample_name, sample_data, participant in zip(sample_names, samples_data, participants):
-            assert isinstance(sample_data['meta'], dict)
-            assert isinstance(sample_data['active'], bool)
-            sample = await self.sl.upsert_sample(
-                SampleUpsertInternal(
-                    external_id=str(sample_data['external_id']),
-                    meta=sample_data['meta'],
-                    type=str(sample_data['type']),
-                    active=sample_data['active'],
-                    participant_id=participant.id,
-                ),
-                project=self.project_id,
-            )
-            self.sample_internal_objects.append(sample)
-            sample_external = sample.to_external()
-            self.sample_external_objects.append(sample_external)
-            setattr(self, f'test_{sample_name}', sample_external)
-            # NOTE: We probably don't need to set the sample_external_objects as attributes of the test class
-            assert sample.id
-
-        self.number_of_samples = len(samples_data)
+        self.sample_internal_objects.extend(
+            sample for participant in participants for sample in participant.samples
+        )
+        self.sample_external_objects.extend(
+            sample.to_external() for sample in self.sample_internal_objects
+        )
 
     @run_as_sync
     async def test_get_dashboard(self):
         """Test get_dashboard"""
         sample_filter = SampleFilter(project=GenericFilter(eq=self.project_id))
-        dashboard = await self.odd.query(sample_filter)
+        dashboard = (await self.odd.query(sample_filter)).to_dict()
```

Comment:
Following on from broken conversation above, not running to_dict here means the static type linting is as useful as running the test.